rendering.js

```diff
@@ -0,0 +1,39 @@
+// __This is the main render function which delegates control to the more-specific render functions.__
+function renderGame(state) {
+    // This would be the mode the game starts in
+    // TODO: It might save us time to start by unifying this game mode state with
+    //  the 'game-over' one so that there's one less code path/state to deal with.
+    if (state.gameMode === 'menu') {
```

Comment:
@poltak I think we should extract all the game modes into a single object like this:
```js
const GAME_MODE = {
  MENU: 'menu',
  PAUSED: 'paused'
  // ...
}

Reply:
yeah good point. I will add that.

I'm used to using TypeScript, where using strings like that is more common as the compiler shouts at you if you get them wrong 😄

Reply:
Should I also be doing this with my gameState in my state mutators repo?

Reply:
Yeah, but maybe after we merge. There's gonna be git conflicts if we both implement the same things, which can be a fun exercise to fix, but I'm sure we'll get more chances to resolve git conflicts 😅